cranelift/codegen/src/isa/aarch64/lower.rs

```diff
@@ -1240,24 +1240,63 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(ctx: &mut C, insn: IRInst) {
             }
         }
 
-        Opcode::Bitrev | Opcode::Clz | Opcode::Cls => {
+        Opcode::Bitrev | Opcode::Clz | Opcode::Cls | Opcode::Ctz => {
+            let has_rev = op == Opcode::Bitrev || op == Opcode::Ctz;
             let rd = output_to_reg(ctx, outputs[0]);
-            let rn = input_to_reg(ctx, inputs[0], NarrowValueMode::None);
-            let op = BitOp::from((op, ty.unwrap()));
-            ctx.emit(Inst::BitRR { rd, rn, op });
-        }
+            let needs_zext = match op {
+                Opcode::Bitrev | Opcode::Ctz => false,
+                Opcode::Clz | Opcode::Cls => true,
+                _ => unreachable!(),
+            };
+            let ty = ty.unwrap();
+            let narrow_mode = if needs_zext && ty_bits(ty) == 64 {
+                NarrowValueMode::ZeroExtend64
+            } else if needs_zext {
+                NarrowValueMode::ZeroExtend32
+            } else {
+                NarrowValueMode::None
+            };
+            let rn = input_to_reg(ctx, inputs[0], narrow_mode);
+            let op_ty = match ty {
+                I8 | I16 | I32 => I32,
+                I64 => I64,
+                _ => panic!("Unsupported type for Bitrev/Clz/Cls"),
+            };
+            let bitop = match op {
+                Opcode::Clz | Opcode::Cls | Opcode::Bitrev => BitOp::from((op, op_ty)),
+                Opcode::Ctz => BitOp::from((Opcode::Bitrev, op_ty)),
+                _ => unreachable!(),
+            };
+            ctx.emit(Inst::BitRR { rd, rn, op: bitop });
+
+            if has_rev {
```

Comment:
nit: Can you replace `has_rev` by its definition here? (since it's used only once)

Reply:
Done!